var/spack/repos/builtin/packages/optional-lite/package.py

```diff
@@ -24,5 +25,39 @@ class OptionalLite(Package):
     version('2.0.0', sha256='e8d803cbc7be241df41a9ab267b525b7941df09747cd5a7deb55f863bd8a4e8d')
     version('1.0.3', sha256='7a2fb0fe20d61d091f6730237add9bab58bc0df1288cb96f3e8a61b859539067')
 
+    def cmake_args(self):
+        return [
+            "-DOPTIONAL_LITE_OPT_BUILD_TESTS=%s"
+            % ("ON" if self.run_tests else "OFF"),
+            "-DOPTIONAL_LITE_OPT_BUILD_EXAMPLES=OFF"
+        ]
+
+    # Pre-3.2.0 install was simply a copytree on the includes
```

Comment:
I think it would make more sense to only override base class methods when they're not in use. So instead of:
```python
def cmake(self, spec, prefix):
    pass

@when("@3.2.0:")
def cmake(self, spec, prefix):
    super(OptionalLite, self).cmake(spec, prefix)
```
you should be able to use:
```python
@when('@:3.1')
def cmake(self, spec, prefix):
    pass
```
Same for other methods and other packages.

Reply:
Good idea! Done.